src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,924 @@
+        ]
+        return downloaded_dekads
+
+    def _get_processed_path(self) -> Path:
+        return self._processed_base_dir / self._get_processed_filename()
+
+    def _get_processed_filename(self) -> str:
+        """Return processed filename.
+
+        Returns the processed filename. The suffix
+        is set using ``self.processed_file_suffix``.
+
+        Returns
+        -------
+        str
+            Processed filename
+        """
+        file_name = (
+            f"{self._country_config.iso3}"
+            f"_usgs_ndvi_{self._data_type}"
+            f"{'_' if len(self.processed_file_suffix) else ''}"
+            f"{self.processed_file_suffix}.csv"
+        )
+        return file_name
+
+    def _get_url(self, filename) -> str:
+        """Get USGS NDVI URL.
+
+        Parameters
+        ----------
+        filename : str
+            File name string generated for specific year, dekad, and data type
+
+        Returns
+        -------
+        str
+            Download URL string
+        """
+        return (
+            f"https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/fews/web/"
+            f"{self._region_url}/dekadal/emodis/ndvi_c6/{self._data_type_url}/"
+            f"downloads/dekadal/{filename}.zip"
+        )
+
+    # TODO: potentially move from static method to
+    # wider USGS function repository
+    @staticmethod
+    def _fp_year_dekad(path: Path) -> List[int]:
+        """Extract year and dekad from filepath.
+
+        Parameters
+        ----------
+        path : Path
+            Filepath
+
+        Returns
+        -------
+        list
+            List of year and dekad
+        """
+        filename = path.stem
+        # find two groups, first for year second for dekad
+        regex = re.compile(r"(\d{2})(\d{2})")
+        return [int(x) for x in regex.findall(filename)[0]]
+
+    def _download_ndvi_dekad(
+        self, year: int, dekad: int, clobber: bool
+    ) -> None:
+        """Download NDVI for specific dekad.
+
+        Parameters
+        ----------
+        year : int
+            Year
+        dekad : int
+            Dekad
+        clobber : bool
+            If True, overwrites existing file
+        """
+        filepath = self._get_raw_path(year=year, dekad=dekad)
+        self._download(filepath=filepath, clobber=clobber)
+
+    # @check_file_existence
+    def _download(self, filepath: Path, clobber: bool):
+        # filepath just necessary for checking file existence
+        # now just extract filepath
+        filename = filepath.stem
+
+        url = self._get_url(filename=filename)
+        try:
+            resp = urlopen(url)
+        except HTTPError:
+            year, dekad = self._fp_year_dekad(filepath)
+            logger.error(
+                f"No NDVI data available for "
+                f"dekad {dekad} of {year}, skipping."
+            )
+            pass
+
+        # open file within memory
+        zf = ZipFile(BytesIO(resp.read()))
```

Comment:
It seems that you can reach this point with `resp` being undefined, because if you get an HTTP error you just log it but don't exit the method

Reply:
Oooh whoops, I had thought `pass` exited the function, but instead have put in an empty return now.